src/main/java/com/artipie/rpm/meta/XmlFilelists.java

```diff
@@ -123,58 +96,26 @@ public XmlFilelists startPackages() throws XMLStreamException {
      */
     public Package startPackage(final String name, final String arch, final String checksum)
         throws XMLStreamException {
-        this.xml.writeStartElement("package");
-        this.xml.writeAttribute("pkgid", checksum);
-        this.xml.writeAttribute("name", name);
-        this.xml.writeAttribute("arch", arch);
-        return new Package(this, this.xml);
+        this.xml.writer().writeStartElement("package");
+        this.xml.writer().writeAttribute("pkgid", checksum);
+        this.xml.writer().writeAttribute("name", name);
+        this.xml.writer().writeAttribute("arch", arch);
+        return new Package(this, this.xml.writer());
     }
 
     @Override
     public void close() throws IOException {
-        final Path trf = Files.createTempFile("filelists-", ".xml");
         try {
-            this.xml.writeEndElement();
-            this.xml.writeEndDocument();
-            this.xml.close();
-            final Transformer transformer = TransformerFactory.newInstance().newTransformer();
-            transformer.setOutputProperty(OutputKeys.VERSION, "1.0");
-            transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
-            transformer.setOutputProperty(OutputKeys.STANDALONE, "no");
-            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
-            final XMLInputFactory factory = XMLInputFactory.newFactory();
-            final XMLEventReader reader = new AlterAttributeEventReader(
-                factory.createXMLEventReader(Files.newInputStream(this.tmp)),
+            this.xml.writer().writeEndElement();
```

Comment:
@victornoel these three lines can be moved to a new method in `XmlFile` called `closeStream()` or something like that to avoid duplication across classes.

Reply:
@victornoel you haven't extracted this functionality

Reply:
@vatavuk I don't know why my answer disappeared, as I was telling, I updated the todo to include this in the task. You can't just extract this from the classes if the matching behaviour to start the xml file stays in here: they both have to happen together. That's what the todo is about. Maybe it is not clear and I can do something to make it more clear? I tried already in my latest commit.

Reply:
@victornoel I don't get why not, we have to close the writer and update a tag. We need them both but they are two separate functionalities. Anyway, let's leave this to the next PR.

Reply:
@vatavuk it is because altering the tag is something separated from creating the original content itself. In this class, creating the original content starts with `startPackages` which do the `writeStartDocument` then `writeStartElement` whose corresponding `writeEndElement` then `writeEndDocument` are the one you commented on. So they must stay together in the same class. The only thing that could actually be moved is `close()` but it would just contain closing of the stream, so there will still be one call here in each of the classes for this `closeStream()` you talk about... so that's what the todo I added is about: factoring all of that and introducing also a `close` method that does more :)